### PR TITLE
Systemd watchdog notification

### PR DESCRIPTION
**Pull request by potaito**
- Notify systemd watchdog while main loop is running
- Disabled alpine Linux CI build, as Alpine is not using systemd

jira: https://auterion.atlassian.net/browse/DIS-4253

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -25,17 +25,17 @@ jobs:
...
-      - uses: actions/checkout@v1
-        with:
-          submodules: recursive
-      - name: install tools
-        run: apk update && apk add autoconf automake build-base git libtool linux-headers openssl-dev perl-extutils-pkgconfig python3 py3-future
-      - name: configure
-        run: ./autogen.sh && ./configure CFLAGS='-g -O2' --sysconfdir=/etc --localstatedir=/var --libdir=/usr/lib --disable-systemd --prefix=/usr
-      - name: build
-        run: make
+  # alpine-linux:
```
Comment (Andreea-L) [inline .github/workflows/build.yml]: If we don't need this block, we should remove it :)

File: src/mavlink-router/mainloop.cpp [hunk 2]
```diff
@@ -290,6 +291,7 @@ int Mainloop::loop()
 
     while (!_should_exit.load(std::memory_order_relaxed)) {
         run_single(-1);
+        sd_notify(0, "WATCHDOG=1");
```
Comment (Andreea-L) [inline src/mavlink-router/mainloop.cpp]: I would add a small comment above this line or above the while, explaining why we need to call notify

**Review by Andreea-L — APPROVED** [review]
LGTM generally, just a couple of small comment :ok_hand:

File: src/mavlink-router/mainloop.cpp (hunk 2 above)
Comment (potaito) [reply src/mavlink-router/mainloop.cpp]: Done, I think it's clearer now

File: .github/workflows/build.yml (hunk 1 above)
Comment (potaito) [reply .github/workflows/build.yml]: Will do, thank you :+1:

**Comment by potaito** [comment]
@Andreea-L 
Per @bkueng's recommendation I modified the watchdog update logic to not update on every single loop iteration (100Hz), but approximately at half of the watchdog threshold.

**Comment by potaito** [comment]
Oh, something's missing. Will rebase in a moment.

**Comment by potaito** [comment]
Done :heavy_check_mark:

**Pull request merged**